doc/man3/DH_new_by_nid.pod

```diff
@@ -24,8 +24,15 @@ B<NID_modp_1536>, B<NID_modp_2048>, B<NID_modp_3072>,
 B<NID_modp_4096>, B<NID_modp_6144> or B<NID_modp_8192>.
 
 DH_get_nid() determines if the parameters contained in B<dh> match
-any named set. It returns the NID corresponding to the matching parameters or
-B<NID_undef> if there is no match. This function is deprecated.
+any named safe prime group. It returns the NID corresponding to the matching
+parameters or B<NID_undef> if there is no match.
+Internally it caches the nid, so that any subsequent calls can fetch the
+cached value.
+Note that if the value of parameter q is not set, then it is set to
+q = (p - 1) / 2.
```

Comment:
It should be clarified that the q is computed only if there is a match for the p:
```
If a matching p is found and if the value of parameter q is not set, then it is set to
q = (p - 1) / 2.
```

Reply:
Updated